examples/scenes/src/cosmic_text_scene.rs

```diff
@@ -0,0 +1,450 @@
+                        }
+                        buffer_line.glyph_highlights.push(Rect::from_origin_size(
+                            Point::new(min as f64, line_top),
+                            Size::new(cmp::max(0, max - min) as f64, line_height),
+                        ));
+                    }
+                }
+            }
+
+            // Cursor
+            if let Some((x, y)) = cursor_position(&editor_info.cursor, &layout_run) {
+                buffer_line.cursor = Some(Rect::from_origin_size(
+                    Point::new(x as f64, y as f64),
+                    Size::new(1.0, line_height),
+                ));
+            }
+        }
+
+        for glyph in layout_run.glyphs {
+            let mut glyph_color = match glyph.color_opt {
+                Some(color) => Color::from_rgba8(color.r(), color.g(), color.b(), color.a()),
+                None => text_color,
+            };
+
+            if let Some(editor_info) = &editor_info {
+                if text_color.components != editor_info.selected_text_color.components {
+                    if let Some((start, end)) = editor_info.selection_bounds {
+                        if line_i >= start.line
+                            && line_i <= end.line
+                            && (start.line != line_i || glyph.end > start.index)
+                            && (end.line != line_i || glyph.start < end.index)
+                        {
+                            glyph_color = editor_info.selected_text_color;
+                        }
+                    }
+                }
+            }
+
+            if let Some((last_font, last_glyph_color)) = last_font {
+                if last_font != glyph.font_id
+                    || last_glyph_color.components != glyph_color.components
+                {
+                    buffer_line.glyph_runs.push(BufferGlyphRun {
+                        font: last_font,
+                        glyphs: current_glyphs,
+                        glyph_color: last_glyph_color,
+                    });
+                    current_glyphs = vec![];
+                }
+            }
+
+            last_font = Some((glyph.font_id, glyph_color));
+            current_glyphs.push(Glyph {
+                x: glyph.x,
+                y: glyph.y + line_y as f32,
+                id: glyph.glyph_id as u32,
+            });
+        }
+        if !current_glyphs.is_empty() {
+            let (last_font, last_color) = last_font.unwrap();
+            buffer_line.glyph_runs.push(BufferGlyphRun {
+                font: last_font,
+                glyphs: current_glyphs,
+                glyph_color: last_color,
+            });
+        }
+
+        buffer_glyphs.buffer_lines.push(buffer_line);
+    }
+
+    buffer_glyphs
+}
+
+// Copied directly from https://github.com/pop-os/cosmic-text/blob/58c2ccd1fb3daf0abc792f9dd52b5766b7125ccd/src/edit/editor.rs#L66
+fn cursor_position(cursor: &Cursor, run: &LayoutRun) -> Option<(i32, i32)> {
```

Comment:
I think this is exposed via https://docs.rs/cosmic-text/latest/cosmic_text/struct.Editor.html#method.cursor_position; I'd much prefer we used that if at all feasible.

Reply:
That function only returns the top left cursor. I think the drawing assumes that there may be multi-line cursors. Although, the editor currently only has one cursor.

Reply:
The reason I want that is for licensing. In the form you've currently included it, we would need to include the full System76 MIT license somewhere (or use this section of code only under the Apache 2.0 license, probably moving it into its own file).